test/python/WMCore_t/Services_t/CRIC_t/CRIC_t.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+"""
+Test case for CRIC
+"""
+from __future__ import print_function
+
+import unittest
+
+from WMCore.Services.CRIC.CRIC import CRIC
+from WMQuality.Emulators.EmulatedUnitTestCase import EmulatedUnitTestCase
+
+class CRICTest(EmulatedUnitTestCase):
+    """
+    Unit tests for SiteScreening module
+    """
+
+    def __init__(self, methodName='runTest'):
+        super(CRICTest, self).__init__(methodName=methodName)
+
+    def setUp(self):
+        """
+        Setup for unit tests
+        """
+        super(CRICTest, self).setUp()
+        self.myCRIC = CRIC()
+
+    def testWhoAmI(self):
+        print("XXXXX")
+        print(self.myCRIC.whoAmI())
+
+    def testUserNameDN(self):
+        """
+        Tests DN to Username lookup
+        """
+        testDn = "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=jha/CN=618566/CN=Manoj Jha"
+        testUserName = "jha"
+        userName = self.myCRIC.userNameDn(testDn)
+        self.assertTrue(testUserName == userName)
+
+    def testDNWithApostrophe(self):
+        """
+        Tests a DN with an apostrophy in - will fail till CRIC2 appears
+        """
+        testDn = "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=liviof/CN=472739/CN=Livio Fano'"
+        testUserName = "liviof"
+        userName = self.myCRIC.userNameDn(testDn)
+        self.assertTrue(testUserName == userName)
+
+    def testPNNstoPSNs(self):
+        """
+        _testPNNstoPSNs_
+
+        Test converting PhEDEx Node Names to Processing Site Names
+        """
+        result = self.myCRIC.PNNstoPSNs(['T1_US_FNAL_Disk', 'T1_US_FNAL_Buffer', 'T1_US_FNAL_MSS'])
+        self.assertTrue(result == ['T1_US_FNAL'])
+        result = self.myCRIC.PNNstoPSNs(['T2_UK_London_IC', 'T2_US_Purdue'])
+        print(result)
+        self.assertItemsEqual(result, [u'T3_UK_London_QMUL', u'T2_US_Purdue', u'T3_UK_SGrid_Oxford',
+                                       u'T2_UK_London_IC', u'T3_UK_ScotGrid_GLA', u'T3_UK_London_RHUL'])
```

Comment:
Alan, 
SiteDB returns only ['T2_UK_London_IC', 'T2_US_Purdue'], But CRIC returns [u'T3_UK_London_QMUL', u'T2_US_Purdue', u'T3_UK_SGrid_Oxford', u'T2_UK_London_IC', u'T3_UK_ScotGrid_GLA', u'T3_UK_London_RHUL']

Do you know why?

Reply:
Are you sure? I just checked the sitedb api and I see those 2 PNNs mapping to a bunch of PSNs (apparently to the same as in CRIC)